File: 2022/01/01.py
calories = []

with open("inputs/01.txt", "r") as file:
    elves  = [[int(item) for item in elf.split()] for elf in file.read().split("\n\n")]

for elf in elves:
    calories.append(sum(elf))

sorted_calories = sorted(calories, reverse=True)

print(f"Part 1: {sorted_calories[0]}")
print(f"Part 2: {sum(sorted_calories[:3])}")

File: 2022/01/01.rb
# frozen_string_literal: true

path = File.expand_path('../inputs/1.txt', __dir__)

calories_per_elf = File.read(path).split("\n\n").map{ |calories| calories.split("\n").map(&:to_i).sum}

part1 = calories_per_elf.max

part2 = calories_per_elf.max(3).sum

puts "part 1:  #{part1} "
puts "part 2:  #{part2} "
